content/docs/tutorials/aws/ec2-webserver.md

```diff
@@ -55,66 +55,85 @@ aws:region: (us-east-1)
 
     ```javascript
     const aws = require("@pulumi/aws");
-
+    
     let size = "t2.micro";     // t2.micro is available in the AWS free tier
-    let ami  = "ami-0ff8a91507f77f867"; // AMI for Amazon Linux in us-east-1 (Virginia)
-
+    
+    let ami = aws.getAmi({
+        filters: [{
+          name: "name",
+          values: ["amzn-ami-hvm-*"],
+        }],
+        owners: ["137112412989"], // This owner ID is Amazon
+        mostRecent: true,
+    });
+    
     let group = new aws.ec2.SecurityGroup("webserver-secgrp", {
         ingress: [
             { protocol: "tcp", fromPort: 22, toPort: 22, cidrBlocks: ["0.0.0.0/0"] },
         ],
     });
-
+    
     let server = new aws.ec2.Instance("webserver-www", {
         instanceType: size,
         securityGroups: [ group.name ], // reference the security group resource above
-        ami: ami,
+        ami: ami.id,
     });
-
+    
     exports.publicIp = server.publicIp;
     exports.publicHostName = server.publicDns;
     ```
 
     ```typescript
-    const aws = require("@pulumi/aws");
-
+    import * as aws from "@pulumi/aws";
+    
     const size = "t2.micro";     // t2.micro is available in the AWS free tier
-    const ami  = "ami-0ff8a91507f77f867"; // AMI for Amazon Linux in us-east-1 (Virginia)
-
+    
+    const ami = aws.getAmi({
+        filters: [{
+            name: "name",
+            values: ["amzn-ami-hvm-*"],
+        }],
+        owners: ["137112412989"], // This owner ID is Amazon
+        mostRecent: true,
+    });
+    
     const group = new aws.ec2.SecurityGroup("webserver-secgrp", {
         ingress: [
             { protocol: "tcp", fromPort: 22, toPort: 22, cidrBlocks: ["0.0.0.0/0"] },
         ],
     });
-
+    
     const server = new aws.ec2.Instance("webserver-www", {
         instanceType: size,
         securityGroups: [ group.name ], // reference the security group resource above
-        ami: ami,
+        ami: ami.id,
     });
-
-    exports.publicIp = server.publicIp;
-    exports.publicHostName = server.publicDns;
+    
+    export const publicIp = server.publicIp;
+    export const publicHostName = server.publicDns;
     ```
 
     ```python
     import pulumi
-    from pulumi_aws import ec2
-    ami = 'ami-0ff8a91507f77f867' # AMI for Amazon Linux in us-east-1 (Virginia)
-
+    import pulumi_aws as aws
```

Comment:
Honestly not sure - but I'll note that we've neraly universally gone with the existing `from pulumi_aws import ec2` style here in other Python examples.  I'm not sure we have to change things here - or should at least agree on what style guideline we're using and consistently change if we believe we are "wrong".

Reply:
So the reason I did this was because I needed the root of the pulumi_aws import. There is no way (AFAICT) to do `from pulumi_aws import ` when it's the root package. So by importing it all, I wanted to add an alias

Reply:
I'm fine with this as is - but FWIW - you could just do `from pulumi_aws import ec2, get_ami`.